pkg/combustion/templates/31-elemental-register.sh.tpl

```diff
@@ -4,12 +4,29 @@ set -euo pipefail
 mkdir -p /etc/elemental
 cp ./{{ .ConfigFile }} /etc/elemental/config.yaml
 
-# Register --no-toolkit disables OS management
-elemental-register --config-path /etc/elemental/config.yaml --state-path /etc/elemental/state.yaml --install --no-toolkit
+# Enable systemd based Elemental registration
+# Register --no-toolkit disables OS management in Rancher
+cat <<- EOF > /etc/systemd/system/elemental-register-systemd.service
+[Unit]
+Description=Elemental Register Install via Systemd
+Wants=network-online.target
+After=network-online.target
+ConditionPathExists=!/etc/rancher/elemental/agent/elemental_connection.json
+
+[Install]
+WantedBy=network-online.target
+
+[Service]
+EnvironmentFile=-/etc/sysconfig/proxy
+Type=oneshot
+ExecStart=/usr/sbin/elemental-register --debug --config-path /etc/elemental/config.yaml --state-path /etc/elemental/state.yaml --install --no-toolkit
+ExecStartPost=/usr/bin/cp /var/lib/elemental/agent/elemental_connection.json /etc/rancher/elemental/agent
+Restart=on-failure
+RestartSec=10
+EOF
 
 # Enable elemental-system-agent
-# On SLEMicro /var/lib is not persistent, so we copy elemental_connection.json in ExecStartPre
-cp /var/lib/elemental/agent/elemental_connection.json /etc/rancher/elemental/agent
+# On SLE Micro /var/lib is not persistent, so we copy elemental_connection.json in ExecStartPre
```

Comment:
```suggestion
```

Reply:
This still needs to persist I think, we don't copy the file during Combustion as this isn't created until post-creation. But the `ExecStartPre` is still there.